Update logspace and bump the version number to 9

tugsbayasgalan opened the pull request:
Summary: This PR adds upgraders for logspace and logspace.out as the optional step size will be deprecated soon. Old models will be using steps size of 100 when nothing is provided.

Differential Revision: D33885098

Note that this PR introduces BC incompatible change since we are deprecating the option of not providing steps size for torch.logspace. If you didn't provide steps in your model, you need to do this to restore old behaviour:

`torch.logspace(start, end) -> torch.logspace(start, end, steps=100)`

cccclai commented on caffe2/serialize/versions.h:
```diff
@@ -58,13 +58,28 @@ constexpr uint64_t kMaxSupportedFileFormatVersion = 0x6L;
+// 2) [01/30/2022]
+//     Bump the version number to 9 to update aten::logspace and
+//     and aten::logspace.out to error out when steps is not
+//     provided. (see: https://github.com/pytorch/pytorch/issues/55951)
```
```suggestion
//     provided. (see: https://github.com/pytorch/pytorch/issues/72051)
```

cccclai commented on test/jit/fixtures_srcs/fixtures_src.py:
```diff
@@ -26,3 +26,19 @@ def __init__(self):
+        super(TestVersionedLogspaceOutV8, self).__init__()
+
+    def forward(self, a: Union[int, float, complex], b: Union[int, float, complex], out: torch.Tensor):
+        return torch.logspace(a, b, out=out)
```
I might miss it earlier. If we are checkout out, we would need to return `out` here, as it will be changed.

tugsbayasgalan committed:
Update logspace and bump the version number to 9 (#72051)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/72051

Test Plan: TestUpgraders.test_aten_logspace && TestSaveLoadForOpVersion.test_aten_logspace

Reviewed By: cccclai

Differential Revision: D33885098

fbshipit-source-id: 68039185a3c1d71834aa60ec425ad84a817084da
